common/search/src/test/scala/weco/api/search/elasticsearch/ElasticsearchServiceTest.scala

```diff
@@ -82,15 +88,176 @@ class ElasticsearchServiceTest
+      val elasticsearchService = new ElasticsearchService(elasticClient)
+      val badIndex = createIndex
+
+      val findByIdFuture = elasticsearchService.findById[ExampleThing](
+        createCanonicalId
+      )(badIndex)
+
+      whenReady(findByIdFuture) { findByIdEither =>
+        val failedFindByID = findByIdEither.left.value
+
+        failedFindByID shouldBe a[IndexNotFoundError]
+      }
+    }
+  }
+
+  describe("findBySearch") {
+    it("finds documents by search") {
+      val thingsToIndex = 0.to(randomInt(5, 10)).map(_ => randomThing).toList
+
+      withExampleIndex(thingsToIndex) { index =>
+        val elasticsearchService = new ElasticsearchService(elasticClient)
+        val thingToQueryFor = thingsToIndex.head
+
+        val searchRequest = searchRequestForThingByName(
+          index = index,
+          name = thingToQueryFor.name
+        )
+
+        val findBySearchFuture =
+          elasticsearchService.findBySearch[ExampleThing](searchRequest)
+
+        whenReady(findBySearchFuture) { findBySearchEither =>
+          val foundBySearch = findBySearchEither.right.value
+          foundBySearch should have size (1)
+          foundBySearch.head shouldBe thingToQueryFor
+        }
+      }
+    }
+
+    it("returns an appropriate error when the specified index does not exist") {
+      val elasticsearchService = new ElasticsearchService(elasticClient)
+      val badIndex = createIndex
+
+      val searchRequest = searchRequestForThingByName(
+        index = badIndex,
+        name = randomAlphanumeric(10)
+      )
+
+      val findBySearchFuture =
+        elasticsearchService.findBySearch[ExampleThing](searchRequest)
+
+      whenReady(findBySearchFuture) { findBySearchEither =>
+        val failedFindBySearch = findBySearchEither.left.value
+        failedFindBySearch shouldBe a[IndexNotFoundError]
+      }
+    }
+  }
+
+  describe("findByMultiSearch") {
+    it("finds documents by performing a MultiSearch") {
+      val thingsToIndex = 0.to(randomInt(5, 10)).map(_ => randomThing).toList
+
+      withExampleIndex(thingsToIndex) { index =>
+        val elasticsearchService = new ElasticsearchService(elasticClient)
+        val thingsToQueryFor = thingsToIndex.slice(0, 3)
+
+        val searchRequests = thingsToQueryFor.map { thing =>
+          searchRequestForThingByName(
+            index = index,
+            name = thing.name
+          )
+        }
+
+        val multiSearchRequest = MultiSearchRequest(searchRequests)
+
+        val findByMultiSearchFuture = elasticsearchService
+          .findByMultiSearch[ExampleThing](multiSearchRequest)
+
+        whenReady(findByMultiSearchFuture) {
+          case (errors, foundBySearch) =>
+            errors should have length (0)
+            foundBySearch.toSet shouldBe thingsToQueryFor.toSet
+        }
+      }
+    }
+
+    it(
+      "returns an appropriate error for only those queries where the specified index does not exist"
```

Comment:
```suggestion
      "returns errors for queries that fail"
```

?

bit less of a mouthful?